File: study-design-patterns/java/factory-pattern/src/product/XmlHandler.java
package product;

public class XmlHandler extends IOHandler{
    @Override
    public void add(String id, String name) {
        System.out.print("XmlHandler add method call!\n");
    }

    @Override
    public void remove(String id) {
        System.out.print("XmlHandler remove method call!\n");
    }

    @Override
    public void update(String id, String name) {
        System.out.print("XmlHandler update method call!\n");
    }

    @Override
    public void query(String id) {
        System.out.print("XmlHandler query method call!\n");
    }
}

File: study-design-patterns/ts/factory-pattern/product/XMLHandler.ts
import IOHandler from "./IOHandler";

export default class XMLHandler extends IOHandler {
  add(): void {
    console.log('XMLHandler add method call!')
  }

  query(): void {
    console.log('XMLHandler query method call!')
  }

  remove(): void {
    console.log('XMLHandler remove method call!')
  }

  update(): void {
    console.log('XMLHandler update method call!')
  }
}